src/newgrf_text.cpp

```diff
@@ -645,6 +638,7 @@ const char *GetGRFStringFromGRFText(const GRFTextWrapper &text)
  */
 const char *GetGRFStringPtr(uint16_t stringid)
 {
+	assert(_grf_text.size() > stringid);
```

Comment:
Doesn't the inverse make more sense? Stringid < size? As I think it should be >= otherwise, not?